Add testing coverage to PRs

jspeed-meyers opened the pull request:
Close #32 - This will help reviewers know if the PR changes code coverage and remind contributors to include tests.

Borrowing code from here: https://github.com/georgetown-cset/funder-finder/blob/main/.github/workflows/main.yml

This GitHub action comes from here: https://github.com/orgoro/coverage

Though the popup box I expect to show up in a PR does not show up here, perhaps it needs to be merged for it to work. My proposal: Merge, see if the PR coverage is working. If not, revert and seek guidance.

Commit:
Add push testing coverage to PR

Close #32 

Borrowing code from here: https://github.com/georgetown-cset/funder-finder/blob/main/.github/workflows/main.yml

goneall reviewed (approved):
Changes looks good to me - with the caveat I'm not a proficient Python programmer.

Commit:
Expand permission to test

Adding an expanded permission could enable testing this feature without a direct merge.

Commit:
Try no permissions as a test

Commit:
Fix yaml typo

jspeed-meyers commented:
Instead of merging and experimenting, I tried more experiments on this branch.

Good news: This PR is working.

Bad news: it is only working with broad permissions enabled, at least right now.

I am going to experiment with limiting permissions and getting this to still work. I don't think it will be hard. I'm just not too experienced with Actions, so it will take a little reading and fiddling.

Commit:
Add some permissions

jspeed-meyers commented:
Tightened the permissions and still working. But I might be able to take it a notch tighter.

Commit:
Comment out actions permissions

Commit:
Comment out contents permission

Commit:
Test downgrading pull-requests permission

Commit:
Use least privilege permissions

This is the least privilege permission that still enables  the code coverage to be sent via PR.

jspeed-meyers commented:
This was the tightest permission level I could create while having the code coverage to PR functionality still work.
